dataprovider/src/main/java/cm/aptoide/pt/dataprovider/ws/v3/OAuth2AuthenticationRequest.java

```diff
@@ -50,7 +51,9 @@ public static OAuth2AuthenticationRequest of(String username, String password, S
           body.put("oauthToken", password);
           break;
         case "FACEBOOK":
-          body.put("authMode", "facebook");
+          String authVal = "facebook";
+          if (cobrand) authVal = "facebook_cobrand";
```

Comment:
I think you didn't need this. I would just put a provider for this string on the flavour application module and then on cobrand you would return "facebook_cobrand" and on vanilla you would return "facebook". Then, instead of injecting a boolean saying if it is cobrand you would inject the authValue that you need to use directly.